packages/@ember/-internals/routing/lib/services/router.ts

```diff
@@ -34,6 +19,17 @@ function cleanURL(url: string, rootURL: string) {
   return url.substr(rootURL.length, url.length);
 }
 
+function getRouter(owner: Owner): EmberRouter | null {
+  try {
+    return owner.lookup('router:main', { instantiate: false }) || null;
+  } catch (error) {
+    if (error.message !== 'Could not create factory') {
+      throw error;
+    }
+    return null;
+  }
+}
+
```

Comment:
@rwjblue For some reason including `{ instantiate:false }` when looking up the router results in a `Could not create factory` error, so for now I wrapped this in a try-catch because I'm not sure of another more proper way of looking up something without attempting to create it.  Is there an alternative or do you think that could be a bug?